atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+        else:
+            dataset = dataset.decode("utf-8") if isinstance(dataset, bytes) else dataset
+        return dataset
+
+    def hdf5_attr_getter(self, source, attribute, path=None, default=None):
+        '''
+        Extracts attribute from object based on path, and returns default if not defined.
+        '''
+        if path:
+            section_segments = path.split('.')
+            for section in section_segments:
+                try:
+                    value = source.get(section)
+                    source = value[-1] if isinstance(value, list) else value
+                except Exception:
+                    return
+        value = source.attrs.get(attribute)
+        source = value[-1] if isinstance(value, list) else value
+        source = self.decode_hdf5_bytes(source, default) if source is not None else default
+
+        return source
+
+    def hdf5_getter(self, source, path, default=None):
+        '''
+        Extracts attribute from object based on path, and returns default if not defined.
+        '''
+        section_segments = path.split('.')
+        for section in section_segments:
+            try:
+                value = source.get(section)
+                unit = self.hdf5_attr_getter(source, 'unit', path=section)
+                unit_factor = self.hdf5_attr_getter(source, 'unit_factor', path=section, default=1.0)
+                source = value[-1] if isinstance(value, list) else value
+            except Exception:
+                return
+
+        if source is None:
+            source = default
+        elif type(source) == h5py.Dataset:
+            source = source[()]
+            source = self.apply_unit(source, unit, unit_factor)
+
+        return self.decode_hdf5_bytes(source, default)
+
+    def parse(self, quantity_key: str = None, **kwargs):
+        pass
+
+
+class H5MDParser(MDParser):
+    def __init__(self):
+        super().__init__()
+        self._data_parser = HDF5Parser()
+        self._n_frames = None  # TODO use trajectory_steps from MDParser?
+        self._n_atoms = None
+        self._frame_rate = None
+        self._atom_parameters = None
+        self._system_info = None
+        self._observable_info = None
+        self._parameter_info = None
+        self._h5md_groups = None
+        self._h5md_particle_group_all = None
+        self._h5md_positions_group_all = None
+        self._h5md_positions_value_all = None
+
+        self._nomad_to_particles_group_map = {
+            'positions': 'position',
+            'velocities': 'velocity',
+            'forces': 'force',
+            'labels': 'species_label',
+            'label': 'force_field_label',
+            'mass': 'mass',
+            'charge': 'charge',
+        }
+
+        self._nomad_to_box_group_map = {
+            'lattice_vectors': 'edges',
+            'periodic': 'boundary',
+            'dimension': 'dimension'
+        }
+
+    @property
+    def h5md_groups(self):
```

Comment:
Do you think it is worthwhile to save these separately? I think we can simply access the group directly from root

Reply:
Do you mean defining a separate property for each group? I thought to do that first, but I wasn't sure which was better.

Reply:
simply having 
```python
self._data_parser.hdf5_getter(self._data_parser.filehdf5, 'particles') since you can not executing each 'group'  more than once so saving the groups does not really offer any benefit.

Reply:
done.